src/map.cpp

```diff
@@ -3341,7 +3341,7 @@ ter_id map::get_roof( const tripoint &p, const bool allow_air )
 
     const auto &ter_there = ter_at( p );
     const auto &roof = ter_there.roof;
-    if( roof.empty() || roof == null_ter_t ) {
+    if( roof.is_null() ) {
```

Comment:
Instead of `roof.is_null()`, you can just use the `operator bool`: `if( root ) { ...` You can use this at other places as well. It's supposed to be similar to the logic of smart pointers.